libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -252,6 +254,31 @@ abstract class DefaultApolloExtension(
     }
   }
 
+  @Suppress("ObjectLiteralToLambda")
+  private fun registerCheckLegacyJsTargetTask(): TaskProvider<Task> {
+    return project.tasks.register(ModelNames.checkLegacyJsTarget()) {
+      val outputFile = BuildDirLayout.legacyJsTargetCheck(project)
+      it.inputs.property("hasLegacyJsTargetTypes", Callable {
+        val kotlin = project.extensions.findByName("kotlin") as? KotlinMultiplatformExtension
+        kotlin?.targets?.any { target -> target is KotlinJsTarget && target.irTarget == null } == true
+      })
+      it.outputs.file(outputFile)
+
+      it.doLast(object : Action<Task> {
+        override fun execute(t: Task) {
+          val hasLegacyJsTargetTypes = it.inputs.properties["hasLegacyJsTargetTypes"] as Boolean
+
+          check(!hasLegacyJsTargetTypes) {
+            "Apollo: LEGACY js target is not supported by Apollo, please use IR or BOTH."
```

Comment:
Maybe just omit "BOTH"? Directly point at IR? 

```suggestion
            "Apollo: LEGACY js target is not supported by Apollo, please use IR."
```

Reply:
Good point 👍